dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -441,19 +496,22 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                         },
                     );
                 }
-            } else if value.ty(&self.body.local_decls, self.tcx).is_integral() {
+            } else if value_ty.is_integral() {
```

Comment:
This isn't new with this PR, but w/r/t this being an `else if`, for code like `*x = myusize as *const _;` (where `x` is an `&mut *const something`), it seems we'll take the `then` branch and emit a `store_value` event but won't emit a `ptr_to_int` event. Do we need to emit the latter in such situations (e.g. to do something with provenance), or do we just care about capturing the pointer value (which the former does)?

Reply:
Good point, although I think it would actually be an `int_to_ptr` event. I'll consider this a bug that should be addressed in another PR.

Reply:
#530